File: python/src/shapes/client/shell/input_reader.py
import sys


def read_input() -> str:
    print("Your Answer >>> ", end="")
    user_input = input()
    user_input = user_input.strip().lower()
    read_exit(user_input)
    return user_input


def read_number() -> float:
    while True:
        user_input = read_input()

        try:
            return float(user_input)
        except ValueError:
            print("Invalid number. Please try again.")


def read_exit(user_input: str):
    if user_input in ["exit", "e"]:
        print("Thank you for using the Shapes CLI!")
        sys.exit(0)

File: scala/src/main/scala/shapes/client/shell/InputReader.scala
package shapes.client.shell

import scala.io.StdIn

private def readInput(): String = {
    print("Your Answer >>> ")
    val userInput = StdIn.readLine().trim.toLowerCase
    readExit(userInput)
    userInput
}

private def readNumber(): Double = {
    while (true) {
        try {
            return readInput().toDouble
        } catch {
            case _: NumberFormatException =>
                println("Invalid number. Please try again.")
        }
    }
    0.0 // Unreachable, but required for compilation
}

private def readExit(userInput: String): Unit = {
    if (userInput == "exit" || userInput == "e") {
        println("Thank you for using the Shapes CLI!")
        sys.exit(0)
    }
}
